config/routes.rb

```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
+  get 'likes/_new'
   root 'users#index'
-  get '/users/:id', to: 'users#show', as: 'user'
-  get '/users/:user_id/posts', to: 'posts#index', as: 'posts_of_user'
-  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'post_of_user'
+
+  resources :users do
+    resources :posts do
+      resources :comments
+      resources :likes
+    end
+  end
```

Comment:
- [ ] It would be better to improve our routing system to match the controller action and only expose the needed routes. Here is how we can fix it
```
resources :users, only: [:index, :show] do
    resources :posts, only: [:new, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create]
    end
  end
```